### ISSUE TITLE
Whitespace collapsed for contents of <pre> element (React/JSX)

### ISSUE THREAD

**Issue by chasm** [dprint/dprint-plugin-typescript#398]
Version: 0.30.3

If I have this:

```jsx
<pre>
  <code className="language-html">
    &lt;p&gt;
      &lt;span&gt;This is some text.&lt;/span&gt;
    &lt;/p&gt;
  </code>
</pre>
```

No matter what I do, if Dprint runs, I get this:

```jsx
<pre>
  <code className="language-html">
    &lt;p&gt; &lt;span&gt;This is some text.&lt;/span&gt; &lt;/p&gt;
  </code>
</pre>
```

Have I done something wrong? Dprint should not be running inside a `<pre>` element, should it? Kind of defeats the purpose. I have a lot of these syntax-highlighted blocks. If I can't turn Dprint off inside them, I pretty much have to give up on it altogether, so I hope I'm missing something.

**Comment by dsherret** [comment]
I also feel like it shouldn't touch anything inside <pre> elements, but it seems prettier also does the same thing as dprint currently does:

https://prettier.io/playground/#N4Igxg9gdgLgprEAuEAzArlMMCW0AEAEnADYkQDqEATiQCYAUAlPsADpT77VwzrWcAPAAceAPg5cugyHTj4wJAIYBnFQDklAWzgBeNiGVQA5uiXG4AWgAWMLSQMTOUrgDISMANzDXxr5Jcpdy8VYSUoXy8AFWscFXw4-BUIHXx4AA8YADpgzwB6UPDIzwCXXLyfPxLnaTzZOCda0QbqgF8QABoQCGFcaBVkUCVqaggAdwAFYYQBlCUSMaUATwGugCNqJTAAa14AZW04ABkcKDhkVHmVOHXNnf2wsFPjZBhqdBuQOC01uDo5OhHcKmcxwABiNC0ShguBMyBASnQMAgnRAtnsFFi8EKYDgexmOFwADdCUt4WA1KjTtdqDAJptjFCLldPgArFTpPbPEhwACK6Ag8GZJGuXTC1Bp8LWSl+JFRolOMAoODoMGsyAAHAAGMWja4UTbCeHNGlE85dACOAvg9J6swRKksZz+f1RPCtOB49PMTKQlxFn2uWhwr3ege5fOt5z9LK6MBlytV6qQACY45scCRngBhFK+r4qACsqPQ1yiMtm-tFICJHwAklA5LA9mBqDhegBBRt7GBLHnC66tVpAA

I think this should be changed.

**Comment by dsherret** [comment]
Thanks for reporting! This should be fixed in 0.71.2 (published once the CI finishes running).

**Comment by chasm** [comment]
Thanks!
